src/http/HttpHeader.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\restful\Http\HttpHeader
+ */
+
+namespace Drupal\restful\Http;
+
+class HttpHeader implements HttpHeaderInterface {
+
+  /**
+   * Header ID.
+   *
+   * @var string
+   */
+  protected $id;
+
+  /**
+   * Header name.
+   *
+   * @var string
+   */
+  protected $name;
+
+  /**
+   * Header values.
+   *
+   * @var array
+   */
+  protected $values;
+
+  /**
+   * Header extras.
+   *
+   * @var string
+   */
+  protected $extras;
+
+  /**
+   * Constructor.
+   */
+  public function __construct($name, array $values, $extras) {
+    $this->name = $name;
+    $this->id = static::generateId($name);
+    $this->values = $values;
+    $this->extras = $extras;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create($key, $value) {
+    list($extras, $values) = self::parseHeaderValue($value);
+    return new static($key, $values, $extras);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function get() {
+    return $this->values;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function contents() {
```

Comment:
maybe ``getString()`` or ``getFormatted()`` (since just by the name it's not clear the difference between ``get()`` and ``contents()``). (btw, Is there a Symfony equivalent - I didn't find one)